ocpi-toolkit-2.2.1/src/main/kotlin/com/izivia/ocpi/toolkit/modules/chargingProfiles/ChargingProfilesCpoServer.kt

```diff
@@ -0,0 +1,75 @@
+package com.izivia.ocpi.toolkit.modules.chargingProfiles
+
+import com.izivia.ocpi.toolkit.common.OcpiSelfRegisteringModuleServer
+import com.izivia.ocpi.toolkit.common.httpResponse
+import com.izivia.ocpi.toolkit.common.mapper
+import com.izivia.ocpi.toolkit.modules.chargingProfiles.domain.SetChargingProfile
+import com.izivia.ocpi.toolkit.modules.versions.domain.InterfaceRole
+import com.izivia.ocpi.toolkit.modules.versions.domain.ModuleID
+import com.izivia.ocpi.toolkit.modules.versions.domain.VersionNumber
+import com.izivia.ocpi.toolkit.modules.versions.repositories.MutableVersionsRepository
+import com.izivia.ocpi.toolkit.transport.TransportServer
+import com.izivia.ocpi.toolkit.transport.domain.HttpMethod
+import com.izivia.ocpi.toolkit.transport.domain.VariablePathSegment
+
+class ChargingProfilesCpoServer(
+    private val service: ChargingProfilesCpoInterface,
+    versionsRepository: MutableVersionsRepository? = null,
+    basePathOverride: String? = null
+) : OcpiSelfRegisteringModuleServer(
+    ocpiVersion = VersionNumber.V2_2_1,
+    moduleID = ModuleID.chargingprofiles,
+    interfaceRole = InterfaceRole.RECEIVER,
+    versionsRepository = versionsRepository,
+    basePathOverride = basePathOverride
+) {
+
+    override suspend fun doRegisterOn(transportServer: TransportServer) {
+        transportServer.handle(
+            method = HttpMethod.GET,
+            path = basePathSegments + listOf(
+                VariablePathSegment("sessionId")
+            )
```

Comment:
I think `queryParams` is missing here:
```suggestion
            method = HttpMethod.GET,
            path = basePathSegments + listOf(
                VariablePathSegment("sessionId")
            ),
            queryParams = listOf("duration", "response_url")
```